src/toil/wdl/wdltoil.py

```diff
@@ -2204,8 +2236,53 @@ def patched_run_invocation(*args: Any, **kwargs: Any) -> List[str]:
+                task_container._run_invocation = patched_run_invocation  # type: ignore
+
+                singularity_original_prepare_mounts = task_container.prepare_mounts
+
+                def patch_prepare_mounts_singularity() -> List[Tuple[str, str, bool]]:
+                    """
+                    Mount the mount points specified from the disk requirements.
 
+                    The singularity and docker patch are separate as they have different function signatures
+                    """
+                    # todo: support AWS EBS/Kubernetes persistent volumes
+                    # this logic likely only works for local clusters as we don't deal with the size of each mount point
+                    mounts: List[Tuple[str, str, bool]] = singularity_original_prepare_mounts()
+                    # todo: support AWS EBS/Kubernetes persistent volumes
+                    # this logic likely only works for local clusters as we don't deal with the size of each mount point
+                    for mount_point, _ in self._mount_spec.items():
+                        abs_mount_point = os.path.abspath(mount_point)
+                        mounts.append((abs_mount_point, abs_mount_point, True))
+                    return mounts
+                task_container.prepare_mounts = patch_prepare_mounts_singularity  # type: ignore[method-assign]
+            elif isinstance(task_container, SwarmContainer):
+                docker_original_prepare_mounts = task_container.prepare_mounts
+
+                try:
+                    # miniwdl depends on docker so this should be available but check just in case
+                    import docker
+                    # docker stubs are still WIP: https://github.com/docker/docker-py/issues/2796
+                    from docker.types import Mount  # type: ignore[import-untyped]
+
+                    def patch_prepare_mounts_docker(logger: logging.Logger) -> List[Mount]:
+                        """
+                        Same as the singularity patch but for docker
+                        """
+                        mounts: List[Mount] = docker_original_prepare_mounts(logger)
+                        for mount_point, _ in self._mount_spec.items():
+                            abs_mount_point = os.path.abspath(mount_point)
+                            mounts.append(
+                                Mount(
+                                    abs_mount_point.rstrip("/").replace("{{", '{{"{{"}}'),
+                                    abs_mount_point.rstrip("/").replace("{{", '{{"{{"}}'),
+                                    type="bind",
+                                )
+                            )
+                        return mounts
+                    task_container.prepare_mounts = patch_prepare_mounts_docker  # type: ignore[method-assign]
+                except ImportError:
+                    logger.warning("Docker package not installed. Unable to add mount points.")
```

Comment:
I'm not sure it makes sense to add the ability to make these mounts in Toil as a monkey-patch. There's nothing here that wouldn't make just as much sense in MiniWDL (or more, since MiniWDL actually has a shared filesystem as an assumption), so instead of doing monkey-patches we should PR this machinery to MiniWDL.

Or if we're starting to add multiple monkey-patches to the TaskContainers, maybe we really want to extend them instead?

Reply:
It probably is a good idea to PR this machinery to MiniWDL. The main reason why I had to monkeypatch this in in the first place is because MiniWDL doesn't actually support the `disks` runtime attribute. Instead of PR-ing the code one-to-one, I think a PR that adds functionality to extend the list of mount points for both docker and singularity is best